packages/grid/x-data-grid/src/components/toolbar/GridToolbarFilterButton.tsx

```diff
@@ -129,8 +137,12 @@ const GridToolbarFilterButton = React.forwardRef<HTMLButtonElement, GridToolbarF
       >
         <rootProps.slots.baseButton
           ref={ref}
+          id={filterButtonId}
           size="small"
           aria-label={apiRef.current.getLocaleText('toolbarFiltersLabel')}
+          aria-controls={isOpen ? filterPanelId : undefined}
```

Comment:
Should the filter header button have same `aria` attributes as it seems to drive the same filter panel?
https://github.com/mui/mui-x/blob/efd46f46d1ec9dca880b8366984539ad1f08ec37/packages/grid/x-data-grid/src/components/columnHeaders/GridColumnHeaderFilterIconButton.tsx#L66-L75